Update the find command to support search by freetime at a specified …

LuoZYi opened the pull request:
…day and update user guide for find and editft command

LuoZYi committed:
Update the find command to support search by freetime at a specified day and update user guide for find and editft command

PohSayKeong commented on src/main/java/seedu/address/logic/parser/FindCommandParser.java:
```diff
@@ -57,11 +62,24 @@ public FindCommand parse(String args) throws ParseException {
+                && argMultimap.getValue(PREFIX_FROM).isPresent()
+                    && argMultimap.getValue(PREFIX_TO).isPresent()) {
+            try {
+                Integer dayOfWeek = Integer.parseInt(argMultimap.getValue(PREFIX_DAY).orElse(null));
```
dont have to orElse here as you have already checked if it is present

PohSayKeong replied:
can use a primitive here

PohSayKeong commented on src/main/java/seedu/address/logic/parser/FindCommandParser.java:
```diff
@@ -57,11 +62,24 @@ public FindCommand parse(String args) throws ParseException {
+                if (dayOfWeek < 1 || dayOfWeek > 5) {
+                    throw new ParseException(EditFreeTimeCommand.MESSAGE_INVALID_DAY);
+                }
+                FreeTime freeTime = ParserUtil.parseFreeTime(argMultimap.getValue(PREFIX_FROM).orElse(null),
```
this is unused

PohSayKeong commented on src/main/java/seedu/address/model/person/predicates/AvailableTimePredicate.java:
```diff
@@ -8,29 +8,43 @@
  * Tests that a {@code Person}'s {@code FreeTime} is within the FreeTime that is given.
  */
 public class AvailableTimePredicate implements FindCommandPredicate {
+    private static final String[] DAY_OF_WEEK = {"Monday", "Tuesday", "Wednesday", "Thursday", "Friday"};
```
Consider using something like this `DayOfWeek.of(i + 1).getDisplayName(TextStyle.SHORT, Locale.ENGLISH);`

PohSayKeong commented on src/main/java/seedu/address/model/person/predicates/AvailableTimePredicate.java:
```diff
@@ -8,29 +8,43 @@
-                    return false;
-                });
+
+        TimeInterval timeInterval = person.getFreeTime().getIntervals().get(day - 1);
```
use `getDay` here

PohSayKeong reviewed (changes requested):
Minor changes required, thanks!

LuoZYi committed:
change as suggested

LuoZYi commented:
Changed as suggested.

PohSayKeong committed:
Update FindCommandParser.java

isPresent check is done therefore there is not need to use orElse.

Updated to use get instead

PohSayKeong reviewed (approved):
LGTM 
Closes #73
